tests/unit_tests/test_config.py

```diff
@@ -147,3 +147,121 @@ def test_auth_from_env_throws_when_not_available():
         BasicAuthentication(username="${BAZ}", password="baz")
     with pytest.raises(KeyError):
         BasicAuthentication(username="${baz}", password="baz")
+
+
+def test_single_dot_notation():
+    """Test with a single dot notation key."""
+    ctx_params = {"BLUEAPI.config.api.host": "my_host"}
+    expected = {"BLUEAPI": {"config": {"api": {"host": "my_host"}}}}
+    result = parse_cli_context(ctx_params)
+    assert result == expected, f"Expected {expected}, but got {result}"
```

Comment:
Doesn't pytest already include expected vs actual context in its exceptions?

Reply:
not sure

Reply:
yes right